vertx-sql-client/src/main/java/io/vertx/sqlclient/spi/Driver.java

```diff
@@ -52,5 +53,32 @@
    * @return true if the driver accepts the {@code connectOptions}, false otherwise
    */
   boolean acceptsOptions(SqlConnectOptions connectOptions);
-
+  
+  /**
+   * @param connectionUri a connection URI
+   * @return true if the driver accepts the {@code uri}, false otherwise
+   */
+  boolean acceptsUri(String connectionUri);
+  
+  /**
+   * Returns a specific {@link SqlConnectOptions} instance corresponding to the {@code uri}.
+   * 
+   * @param <T> the expected type of {@code SqlConnectOptions}
+   * @param connectionUri a connection URI
+   * @return connect options specific to the driver
+   * @throws IllegalArgumentException if the driver does not accept the {@code uri}
+   */
+  <T extends SqlConnectOptions> T getConnectOptions(String connectionUri);
```

Comment:
I can't see where this is used in this PR

Reply:
Hi,

In order to create a `Pool` or a `ConnectionFactory` using `createPool()` or `createConnectionFactory()`, we basically need to be able to create specific `SqlConnectOptions` (eg. `PgConnectOptions`). So the normal flow to use a `Driver` would be:

```java
Vertx vertx = Vertx.vertx();

// From configuration file
String connectionUri = ...
JsonObject dbConnectOptionsJson = ...

// Get instance corresponding to the connection URI using DriverLoader
Driver driver = DriverLoader.getDriver(connectionUri);

// Load specific Sql connect options
SqlConnectOptions dbConnectOptions = optionsdriver.getConnectOptions(connectionUri, dbConnectOptions);

// Get specific connection factory
ConnectionFactory connectionFactory = driver.createConnectionFactory(vertx, dbConnectOptions);
```
That way we can create a code that only depends on vertx-sql-client and which can create a `Pool`or a `ConnectionFactory` for a specific database assuming the proper Driver is on the classpath at runtime (eg. vertx-pg-client).

I hope this clarifies, this should go to unit tests by the way :-)

Reply:
users should not interact with the driver class directly

Reply:
only frameworks are intended to be